hivemq-edge/src/frontend/src/modules/ProtocolAdapters/hooks/useGetAdapterInfo.ts

```diff
@@ -0,0 +1,27 @@
+import { useMemo } from 'react'
+import { ProtocolAdapter } from '@/api/__generated__'
+import { useGetAdapterTypes } from '@/api/hooks/useProtocolAdapters/useGetAdapterTypes.ts'
+import { useListProtocolAdapters } from '@/api/hooks/useProtocolAdapters/useListProtocolAdapters.ts'
+
+const useGetAdapterInfo = (adapterId: string | undefined) => {
+  const { data: allProtocols, isLoading: isProtocolLoading } = useGetAdapterTypes()
+  const { data: allAdapters, isLoading: isAdapterLoading } = useListProtocolAdapters()
+
+  const { adapter, isDiscoverable, protocol } = useMemo(() => {
+    const adapter = allAdapters?.find((e) => e.id === adapterId)
+    const protocol: ProtocolAdapter | undefined = allProtocols?.items?.find((e) => e.id === adapter?.type)
```

Comment:
suggestion(non-blocking): found 2 more 

```suggestion
    const adapter = allAdapters?.find((adapter) => adapter.id === adapterId)
    const protocol: ProtocolAdapter | undefined = allProtocols?.items?.find((protocol) => protocol.id === adapter?.type)
```